homeassistant/components/zwave_js/helpers.py

```diff
@@ -0,0 +1,92 @@
+"""Helper functions for Z-Wave JS integration."""
+from typing import List, Tuple, cast
+
+from zwave_js_server.client import Client as ZwaveClient
+from zwave_js_server.model.node import Node as ZwaveNode
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import async_get as async_get_dev_reg
+from homeassistant.helpers.entity_registry import async_get as async_get_ent_reg
+
+from .const import DATA_CLIENT, DOMAIN
+
+
+@callback
+def get_device_id(client: ZwaveClient, node: ZwaveNode) -> Tuple[str, str]:
+    """Get device registry identifier for Z-Wave node."""
+    return (DOMAIN, f"{client.driver.controller.home_id}-{node.node_id}")
+
+
+@callback
+def get_home_and_node_id_from_device_id(device_id: Tuple[str, str]) -> List[str]:
+    """
+    Get home ID and node ID for Z-Wave device registry entry.
+
+    Returns [home_id, node_id]
+    """
+    return device_id[1].split("-")
+
+
+@callback
+def async_get_node_from_device_id(hass: HomeAssistant, device_id: str) -> ZwaveNode:
+    """Get node from a device ID."""
```

Comment:
I think we should add to the docstring body what exceptions are raised. We only want to leak exceptions inside service handlers. If this helper is used elsewhere we need to catch those exceptions.